src/box/sql/mem.c

```diff
@@ -2943,6 +2943,17 @@ port_c_get_vdbemem(struct port *base, uint32_t *size)
 			if (mem_copy_str(&val[i], str, len) != 0)
 				goto error;
 			break;
+		case MP_BIN:
+			str = mp_decode_bin(&data, &len);
+			if (mem_copy_bin(&val[i], str, len) != 0)
+				goto error;
+			break;
+		case MP_EXT:
+			str = data;
+			mp_next(&data);
```

Comment:
Is this next required to skip mp_ext header containing type of extension?

Reply:
Yes. This mp_next moves data to the start of next value in mp_array. It does not matter which mp_type current value has. In case of mp_ext it skips header of the value (MP_EXT) + header of extension (MP_EXT_...) + length of the value.